src/librustc_mir/transform/promote_consts.rs

```diff
@@ -266,8 +277,10 @@ impl<'a, 'tcx> Promoter<'a, 'tcx> {
         });
 
         // Inject the Rvalue or Call into the promoted MIR.
+        let dest = Lvalue::Temp(new_temp);
+        self.push(StatementKind::StorageLive(dest.clone()), source_info.span);
```

Comment:
Note that LLVM doesn't require matching intrinsic calls, so there's no strong reason to require matching `Storage{Live,Dead}` statements in MIR. It just considers the location dead before the `lifetime.start` call and after the `lifetime.end` call.